test/unit/assorted/optional_require.test.ts

```diff
@@ -0,0 +1,72 @@
+import { expect } from 'chai';
+import { existsSync } from 'fs';
+import { resolve } from 'path';
+
+import {
+  AuthContext,
+  compress,
+  GSSAPI,
+  HostAddress,
+  MongoDBAWS,
+  MongoMissingDependencyError
+} from '../../mongodb';
+
+function moduleExistsSync(moduleName) {
```

Comment:
Changed this file to .ts and migrated callback code to promises. callbacks were just ignored so no one was handling the outcome of the promises.